README.md

```diff
@@ -1,2 +1,186 @@
+  the window position
+* If neither my previous route nor current route define a `scrollFrame`, reset
+  the window position
+* If the same route was clicked twice in a row and the query and search didn't
+  change, reset the window position
+* else do nothing
+
+## API Reference
+
+`react-simple-scroll` exports a component and a router middleware factory:
+
+```jsx
+import SimpleScroll, { scrollMiddleware } from 'react-simple-scroll'
+```
+
+#### `scrollMiddleware(props)`
+
+`scrollMiddleware` is a function that accepts props to bind to the underlying
+`<SimpleScroll />` component and returns an appropriate middleware for
+`react-router`.
+
+```jsx
+import { scrollMiddleware } from 'react-simple-scroll';
+import isEqual from 'lodash.isequal'
+
+
+const middleware = scrollMiddleware({ isEqual });
+```
+
+#### `<SimpleScroll />`
+
+This component will likely never be used directly by the user.  It emits no DOM
+and is designed to sit between the `<Router />` and `<RouterContext />`
+components of your heirarchy.
+
+
+##### props
+
+| Name                              | Type   | Required? | Description                           |
+|-----------------------------------|--------|-----------|---------------------------------------|
+| routerProps                       | object | true      | Supplied by react-router              |
+| isEqual                           | func   | true      | Returns true if two objects are equal |
+| enableBrowserScrollRestoration    | bool   | false     | Default `false`, see [Scroll Restoration](#scroll-restoration)             |
+| children                          | node   | false     | Supplied by react-router              |
+
+## Scroll Restoration
+
+Many history implementations for react-router will fall back to the
+[History API]() provided by most modern browsers.  The History API has a feature
+wherin scroll positions are recorded when pushing and restored when popping
+pages.  This has the unwanted side-effect of hijacking the scrolling we're
+trying to manually set here.  By default, `react-simple-scroll` will make an
+effort to disable this feature.  If however you'd rather leave it enabled,
+simply pass the `enableBrowserScrollRestoration` to `scrollMiddleware`:
+
+```jsx
+const middleware = scrollMiddleware({
+	isEqual,
+	enableBrowserScrollRestoration: true
+});
+```
+
+## License
+
+MIT
+
+## Contributing
+
+If you're interested in contributing to `react-simple-scroll`, a good place to
+start is by opening up an
+[Issue](https://github.com/button/react-simple-scroll/issues) and describing the
+change you'd like to see, be it a bug, feature request, or otherwise.  This
+gives everyone a chance to review the proposal from a high-level before any
+development effort is invested.
+
+#### Lifecycle of a Change
+
+* Open an [Issue](https://github.com/button/react-simple-scroll/issues) describing the change
+* Create a new branch for your changes: `git checkout -b <user>/update-bloop`
```

Comment:
Might wanna make "fork repo" an explicit step, could see people who aren't common OS contributors not knowing